src/LayerPlanBuffer.h

```diff
@@ -54,38 +60,63 @@ class LayerPlanBuffer : SettingsMessenger
     {
         preheat_config.setConfig(settings);
     }
-    
+
     /*!
-     * Place a new layer plan (GcodePlanner) by constructing it with the given arguments.
-     * Pop back the oldest layer plan is it exceeds the buffer size and write it to gcode.
+     * Push a new layer plan into the buffer
      */
-    template<typename... Args>
-    GCodePlanner& emplace_back(Args&&... constructor_args)
+    void push(LayerPlan& layer_plan)
     {
+        buffer.push_back(&layer_plan);
+    }
+
+    /*!
+     * Process all layers in the buffer
+     * This inserts the temperature commands to start warming for a given layer in earlier layers
```

Comment:
done

The spirit is 'Process all layers in the buffer'. I could expand on that.....
I do want to be explicit about what it exactly does, though, because it might come as a surprise.
Fan speeds and travel moves are more logically located in LayerPlan, but are not possible there due to multithreading.